### ISSUE TITLE
Problem with multiple calls to hasMember

### ISSUE THREAD

**Issue by pdbreen** [spatie/laravel-newsletter#38]
When making successive calls to hasMember (v3.0.2), I'm finding that the result will get stuck on false after the first proper false response.

I believe the problem has to do with the `lastActionSucceeded()` check after the `lists->findByName()` within getMember.  Since no new API call was made, this check is out of place and returning the result of the prior API call which is no longer relevant to the current request.

```
    public function getMember($email, $listName = '')
    {
        $list = $this->lists->findByName($listName);

        if (!$this->lastActionSucceeded()) {
            return false;
        }
```

**Comment by freekmurze** [comment]
Maybe related to #36 

Please try upgrading to the latest version of this package and the underlying mailchimp package.

Reopen this issue if the problem persists.

**Comment by pdbreen** [comment]
Nope - not related (and I've pinned underlying lib to 2.2 because of that issue).

The problem is that you're checking for an API failure when no (new) API call has been made, so you're not getting a valid result.

**Comment by drbyte** [comment]
I'm having (what I think is) the same problem. In this test it's with Release 3.0.4 on L 5.2, PHP 7.10-RC1

```sh
php artisan tinker

use Spatie\Newsletter\NewsletterFacade as Newsletter;
=> null

Newsletter::hasMember('foo@bar.com');
=> false

Newsletter::subscribe('foo@bar.com');
{tinker crashes}
```

```sh
php artisan tinker

use Spatie\Newsletter\NewsletterFacade as Newsletter;
=> null

Newsletter::subscribe('foo@bar.com');
=> lots of data output

Newsletter::subscribe('foo@bar.com');
{tinker crashes}
```

```sh
php artisan tinker

use Spatie\Newsletter\NewsletterFacade as Newsletter;
=> null

Newsletter::hasMember('foo@bar.com');
=> true

Newsletter::hasMember('foo@bar.com');
{tinker crashes}
```

**Comment by freekmurze** [comment]
I'll fix this when I have a little more time. I'd accept a PR that fixes this.

**Comment by freekmurze** [comment]
Fixed in `3.0.5`
